app/Http/Middleware/CheckRegistrationsAllowed.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace App\Http\Middleware;
+
+use Closure;
+use Illuminate\Http\Request;
+
+class CheckRegistrationsAllowed
+{
+    /**
+     * Handle an incoming request.
+     *
+     * @param Request $request
+     * @param Closure $next
+     * @return mixed
+     */
+    public function handle($request, Closure $next)
+    {
+        if (!config('auth.misc.users.allow_registration')) {
+            return redirect('/login');
```

Comment:
Would we be able to just throw a 404 or similar error instead - that way it is more clear theres a problem rather than a redirect without any reasoning?

Reply:
sure